From 5d4d0d153b43b5dcf8f456aa1134daae2c3010e7 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 8 Mar 2015 19:54:56 +0200
Subject: [PATCH 0141/1240] spl: a8k: added support for SPL for A8k SoC

	- Added support for SPL for A8k SoC
	- Set the TEXT base of the SPL to 0x0

Change-Id: Ifc239b4d19dc215302845ec926fb384c30a46ab6
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/17315
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
---
 arch/arm/Kconfig                 |  1 +
 configs/mvebu_armada8k_defconfig |  8 +++++---
 include/configs/armada8k.h       | 13 +++++++++++++
 3 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 49ba49a..4626c8c 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -104,6 +104,7 @@ config TARGET_VCMA9
 
 config TARGET_ARMADA_8K
 	bool "Support Armada 8K"
+	select SUPPORT_SPL
 
 config TARGET_ARMADA_38X
 	bool "Support Armada 38x"
diff --git a/configs/mvebu_armada8k_defconfig b/configs/mvebu_armada8k_defconfig
index 52ab222..8024c3a 100644
--- a/configs/mvebu_armada8k_defconfig
+++ b/configs/mvebu_armada8k_defconfig
@@ -1,7 +1,9 @@
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
-CONFIG_ARM=y
-CONFIG_TARGET_ARMADA_8K=y
-CONFIG_MVEBU_UART=y
+CONFIG_SPL=y
++S:CONFIG_ARM=y
++S:CONFIG_TARGET_ARMADA_8K=y
++S:CONFIG_MVEBU_UART=y
++S:CONFIG_MVEBU_MK6_SUPPORT=y
 CONFIG_ARMV8_PSCI=y
 CONFIG_CMD_BDI
 CONFIG_CMD_BOOTD
diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 197c415..1073f5e 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -49,6 +49,19 @@
 #define COUNTER_FREQUENCY	(46300)
 #define CONFIG_MSS_FREQUENCY	(125000)
 
+/* SPL */
+/* Defines for SPL */
+#define CONFIG_SPL_TEXT_BASE		0x0
+#define CONFIG_SPL_MAX_SIZE		(0x1ffc0)
+
+#define CONFIG_SPL_BSS_START_ADDR	(0x20000)
+#define CONFIG_SPL_BSS_MAX_SIZE		(0x8000)
+
+#define CONFIG_SYS_SPL_MALLOC_START	(CONFIG_SPL_BSS_START_ADDR + \
+					 CONFIG_SPL_BSS_MAX_SIZE)
+#define CONFIG_SYS_SPL_MALLOC_SIZE	(0x4000)
+#define CONFIG_SPL_LDSCRIPT		"arch/arm/cpu/mvebu-common/u-boot-armv8-spl.lds"
+
 /*
  * Include the generic MVEBU configuration file.
  * The generic file is affected by the defines above.
-- 
1.9.1

